### PR TITLE
 dotCMS/core#17606 SPA: Set active state in nav items

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: components/layout/Nav/ItemLink.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+            activeClassName="active"
+            className="rd-nav-item"
             to={{
                 pathname: props.pathname,
                 state: props.state
             }}
         >
-            {props.children}
-        </Link>
+            <span className={props.className}>{props.children}</span>
```
Comment (fmontes) [inline components/layout/Nav/ItemLink.js]: Can we pass this class in the `NavLink`?

File: components/layout/Nav/NavOption.js [hunk 2]
```diff
@@ -4,11 +4,11 @@ import ItemLink from './ItemLink';
 
 const NavOption = ({ item }) => {
     return (
-        <li className="rd-nav-item" key={item.folder}>
-            <ItemLink className="rd-nav-link" pathname={item.href}>
+        <ItemLink className="rd-nav-link" pathname={item.href}>
```
Comment (fmontes) [inline components/layout/Nav/NavOption.js]: 🤯🤯🤯🤯🤯🤯🤯🤯

File: components/layout/Nav/ItemLink.js (hunk 1 above)
Comment (alfredo-dotcms) [reply components/layout/Nav/ItemLink.js]: nope, otherwise styles will break

File: components/layout/Nav/ItemLink.js [hunk 3]
```diff
@@ -1,16 +1,53 @@
...
+            )}
+        </li>
+    );
+};
+
+const ItemLink = parentProps => {
+    return (
+        <NavLink
+            component={props => {
+                return parentProps.topMenu ? getListItem(parentProps, props) : getLink(parentProps, props);
```
Comment (fmontes) [inline components/layout/Nav/ItemLink.js]: In react, this should be components

```suggestion
                return parentProps.topMenu ? <ListItem {...parentProps, ...props} /> : <Link {...parentProps, ...props} />;
```

File: components/layout/Nav/ItemLink.js [hunk 4]
```diff
@@ -1,16 +1,53 @@
...
+            {!Array.isArray(parentProps.children) ? (
+                getLink(parentProps, props)
+            ) : (
+                <>{parentProps.children}</>
+            )}
+        </li>
+    );
+};
+
+const ItemLink = parentProps => {
```
Comment (fmontes) [inline components/layout/Nav/ItemLink.js]: Now that this does so many different things, should we still calling `ItemLink`? should be break this down somehow?

File: components/layout/Nav/ItemLink.js [hunk 5]
```diff
@@ -1,16 +1,53 @@
...
+            onMouseOut={() => {
+                setFocus(false);
             }}
         >
-            {props.children}
-        </Link>
+            {!Array.isArray(parentProps.children) ? (
+                getLink(parentProps, props)
+            ) : (
+                <>{parentProps.children}</>
```
Comment (fmontes) [inline components/layout/Nav/ItemLink.js]: Why do you need the fragment here? it could be more than one children?

**Pull request merged**